tools/src/misc/CMakeLists.txt

```diff
@@ -5,122 +5,88 @@ project (HDF5_TOOLS_SRC_MISC C)
+set_target_properties (h5delete PROPERTIES FOLDER tools)
+set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5delete")
 
-  add_executable (h5repart-shared ${HDF5_TOOLS_SRC_MISC_SOURCE_DIR}/h5repart.c)
-  target_include_directories (h5repart-shared PRIVATE "${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  target_compile_options(h5repart-shared PRIVATE "${HDF5_CMAKE_C_FLAGS}")
-  TARGET_C_PROPERTIES (h5repart-shared SHARED)
-  target_link_libraries (h5repart-shared PRIVATE ${HDF5_TOOLS_LIBSH_TARGET} ${HDF5_LIBSH_TARGET})
-  set_target_properties (h5repart-shared PROPERTIES FOLDER tools)
-  set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5repart-shared")
-
-  add_executable (h5mkgrp-shared ${HDF5_TOOLS_SRC_MISC_SOURCE_DIR}/h5mkgrp.c)
-  target_include_directories (h5mkgrp-shared PRIVATE "${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  target_compile_options(h5mkgrp-shared PRIVATE "${HDF5_CMAKE_C_FLAGS}")
-  TARGET_C_PROPERTIES (h5mkgrp-shared SHARED)
-  target_link_libraries (h5mkgrp-shared PRIVATE ${HDF5_TOOLS_LIBSH_TARGET} ${HDF5_LIBSH_TARGET})
-  set_target_properties (h5mkgrp-shared PROPERTIES FOLDER tools)
-  set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5mkgrp-shared")
-
-  add_executable (h5clear-shared ${HDF5_TOOLS_SRC_MISC_SOURCE_DIR}/h5clear.c)
-  target_include_directories (h5clear-shared PRIVATE "${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  target_compile_options(h5clear-shared PRIVATE "${HDF5_CMAKE_C_FLAGS}")
-  TARGET_C_PROPERTIES (h5clear-shared SHARED)
-  target_link_libraries (h5clear-shared PRIVATE ${HDF5_TOOLS_LIBSH_TARGET} ${HDF5_LIBSH_TARGET})
-  set_target_properties (h5clear-shared PROPERTIES FOLDER tools)
-  set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5clear-shared")
-
-  add_executable (h5delete-shared ${HDF5_TOOLS_SRC_MISC_SOURCE_DIR}/h5delete.c)
-  target_include_directories (h5delete-shared PRIVATE "${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  target_compile_options(h5delete-shared PRIVATE "${HDF5_CMAKE_C_FLAGS}")
-  TARGET_C_PROPERTIES (h5delete-shared SHARED)
-  target_link_libraries (h5delete-shared PRIVATE ${HDF5_TOOLS_LIBSH_TARGET} ${HDF5_LIBSH_TARGET})
-  set_target_properties (h5delete-shared PROPERTIES FOLDER tools)
-  set_global_variable (HDF5_UTILS_TO_EXPORT "${HDF5_UTILS_TO_EXPORT};h5delete-shared")
-
-  set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES}
-      h5debug-shared
-      h5repart-shared
-      h5mkgrp-shared
-      h5clear-shared
-      h5delete-shared
-  )
-endif ()
+set (H5_DEP_EXECUTABLES
```

Comment:
In the static block, I see:

```
  set (H5_DEP_EXECUTABLES
      h5debug
      h5repart
      h5mkgrp
      h5clear
      h5delete
  )
```

but in the shared block I see:

```
  set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES}
      h5debug-shared
      h5repart-shared
      h5mkgrp-shared
      h5clear-shared
      h5delete-shared
  )
```

Should here and everywhere else that's setting `H5_DEP_EXECUTABLES` actually be using the latter version to append to the list rather than overwriting it?

Reply:
That should have been deleted.

Reply:
> That should have been deleted.

Do you mean `H5_DEP_EXECUTABLES` should have been deleted everywhere? Otherwise, the old static block is deleted in your changes, but where it gets set throughout different CMake files alternates between `set (H5_DEP_EXECUTABLES ...)` and `set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES} ...`. Just wondering if it's always supposed to be `set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES} ...` everywhere.

Reply:
No the variable is reset for each tool

Reply:
> > That should have been deleted.
> 
> Do you mean `H5_DEP_EXECUTABLES` should have been deleted everywhere? Otherwise, the old static block is deleted in your changes, but where it gets set throughout different CMake files alternates between `set (H5_DEP_EXECUTABLES ...)` and `set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES} ...)`. Just wondering if it's always supposed to be `set (H5_DEP_EXECUTABLES ${H5_DEP_EXECUTABLES} ...)` everywhere.

Sorry - I thought you were referring to code that was still in the file.

Reply:
Oh I see, it's just collecting a list of targets for the `install` command and some files only need to set it once. That file was previously appending to the list in case both static and shared tools were built. Makes sense.